build-tools/packages/version-tools/src/internalVersionScheme.ts

```diff
@@ -108,50 +108,78 @@ export function toInternalScheme(
 }
 
 /**
- * Validates that the version follows the Fluid internal version scheme. Throws if not.
+ * Validates that the version follows the Fluid internal version scheme.
+ *
+ * @param version - The version to check.
+ * @param allowPrereleases - If true, allow prerelease Fluid internal versions.
+ * @returns True if the version matches the Fluid internal version scheme. Throws if not.
+ *
+ * @remarks
+ *
+ * This function is not typically used. {@link isInternalVersionScheme} is more useful since it does not throw.
  */
 // eslint-disable-next-line @rushstack/no-new-null
-function validateVersionScheme(version: semver.SemVer | string | null) {
+function validateVersionScheme(version: semver.SemVer | string | null, allowPrereleases = false) {
     const parsedVersion = semver.parse(version);
     if (parsedVersion === null) {
         throw new Error(`Couldn't parse ${version} as a semver.`);
     }
 
-    if (parsedVersion.prerelease.length !== 4) {
-        throw new Error(
-            `Prerelease value doesn't contain 4 components; found ${parsedVersion.prerelease.length}`,
-        );
-    }
-
     if (parsedVersion.prerelease[0] !== "internal") {
         throw new Error(
             `First prerelease component should be internal; found ${parsedVersion.prerelease[0]}`,
         );
     }
 
     if (parsedVersion.major < 2) {
-        throw new Error(`The public major version must by >= 2; found ${parsedVersion.major}`);
+        throw new Error(`The public major version must be >= 2; found ${parsedVersion.major}`);
+    }
+
+    if (parsedVersion.prerelease.length > 4) {
```

Comment:
Might be worth adding a comment explaining where the 4 comes from here / extracting it into a constant.